### PR TITLE
Fixes #792 and Fixes #811 with docs overhaul

### LINKED ISSUE #792: Document new scoping and functions - big docs overhaul.
Everything that happened in 0.17.0 needs work in the docs - this issue is specifically for the user docs for user functions, scoping keywords, and such (issue #750)

### LINKED ISSUE #811: [can't be fixed][documentation] omitting parentheses in function call only works if a function is defined in the same file
I know that it can't be fixed.
What we can do about it:
1. either document it
2. or don't tell our users that they can omit  parentheses
```
// library.ks
function foo
{
  return 42.
}
```
```
// main.ks
run library.
print foo. // error
```
```
// main.ks
run library.
print foo(). // fine
```
```
// main.ks
function foo
{
  return 42.
}
print foo. // fine
```

### PR DESCRIPTION
**Pull request by Dunbaratu**
S...Soooo much..... typing......

### CODE REVIEW

File: doc/source/language/user_functions.rst [hunk 1]
```diff
@@ -37,7 +38,27 @@ Help for the new user - What is a Function?
 In kerboscript, you can make your own user functions using the
 DECLARE FUNCTION command, which has syntax as follows:
 
-  ``declare function`` *identifier* ``{`` *statements* ``}`` *optional dot (.)*
+  [``declare``] [``local``|``global``] ``function`` *identifier* ``{`` *statements* ``}`` *optional dot (.)*
```
Comment (abenkovskii) [inline doc/source/language/user_functions.rst]: when I typed `global function` it sad that `global` only works with variables

Comment (Dunbaratu) [inline doc/source/language/user_functions.rst]: Okay I'll change it tomorrow morning.

File: doc/source/language/user_functions.rst [hunk 2]
```diff
@@ -349,24 +468,22 @@ declaring them first.  Most such languages have provided a way to
 catch the problem, and allow you to instruct the compiler "please
 don't let me do that.  Please force me to declare everything".
 
-The way that is done in kerboscript is by using a ``NOLAZYGLOBAL`` 
-section, :ref:`as described here <nolazyglobal>`.
+The way that is done in kerboscript is by using a ``@LAZYGLOBAL`` 
+compiler directive, :ref:`as described here <lazyglobal>`.
 
-Had the function above been wrapped inside a NOLAZYGLOBAL section,
+Had the function above been wrapped inside a ``@LAZYGLOBAL off.`` section,
```
Comment (teleksterling) [inline doc/source/language/user_functions.rst]: It's not technically 'wrapped' in a ``@LAZYGLOBAL off.`` section, right? It's for the whole file.

Comment (Dunbaratu) [inline doc/source/language/user_functions.rst]: good catch.  I thought I found all of those and changed them.  I'll issue a PR to change it.

**Pull request merged**